strawberryfields/tdm/tdmprogram.py

```diff
@@ -282,6 +310,121 @@ def context(self, *args, copies=1, shift="default"):
         self.loop_vars = self.params(*[f"p{i}" for i in range(len(args))])
         return self
 
+    def compile(self, *, device=None, compiler=None):
+        """Compile the time-domain program given a Strawberry Fields photonic hardware device specification.
+        At this stage the compilation is simply a check that the program matches the device.
+
+        Args:
+            device (~strawberryfields.api.DeviceSpec): device specification object to use for
+                program compilation
+            compiler (str, ~strawberryfields.compilers.Compiler): Compiler name or compile strategy
+                to use. If a device is specified, this overrides the compile strategy specified by
+                the hardware :class:`~.DeviceSpec`. For :class:`~.TDMProgram` this should be set to
+                None.
+
+        Returns:
+            Program: compiled program
+        """
+        if compiler == "gaussian":
+            return super().compile(device=device, compiler=compiler)
+
+        if compiler == "tdm_one_loop" and device == "tdm_one_loop_device":
+            # Do something clever
+            compiler = "TD2"  # Devious but it will work, Josh said
+            # Do not add a return statement
```

Comment:
🙈

Reply:
I'm struggling to recall the point of this if statement @nquesada 

* `device` cannot be a string, yet it is being compared to a string. So this if statement will never be activated.

* Even if the `if` statement _is_ activated, the internal logic doesn't seem to do anything.

Reply:
Hey @josh146 --- this is partly my fault. I still need to add the logic there to some "clever" transformations. I left those comments there just a reminder for myself. They can be removed and I can the later add a branch with the actual logic.

Reply:
no worries @nquesada, I was worried that I was forgetting my own devious reasoning here 😆